### DISCUSSION TITLE
Quick question on aborting firebase calls

### DISCUSSION THREAD

**Discussion by misfityudi** [invertase/react-native-firebase#6723] [Q&A]
In the app that I'm working on, I'm making most of the network calls via rnfirebase. As you would know, we do have to clean up useEffects to prevent memory leaks, I'm really struggling to do so. I've looked a lot for documentation on aborting network requests as they do via abort controller for Fetch and Axios calls, but I've been unable to find anything.

I would really appreciate it if someone could help, maybe link me to a relevant doc.

**Comment by mikehardy** [comment]
Hi there - I've never investigated aborting network calls personally, and I'm not aware of any documentation around the same.
I don't believe there is any way to stop an in flight call to any of the firebase APIs, though it would be interesting to see an example of which calls in which module you were concerned about. The API documentation you see at rnfirebase.io is what we have though - nothing more or less and there's no abort control channel in any of them either at our level (react-native-firebase) or at the SDK level we wrap (firebase-ios-sdk / firebase-android-sdk)
